app/main.py

```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
```

Comment:
it's better to use `else` here to avoid duplicating `return self`

Reply:
> it's better to use `else` here to avoid duplicating `return self`

Ok, 'else' added